### ISSUE TITLE
[BUG] Wrong Accept-Language sent with Firefox only

### ISSUE THREAD

**Issue by lazka** [microsoft/playwright#23732]
### System info
- Playwright Version: 1.32.3
- Operating System: Ubuntu 23.04
- Browser: Firefox
- Other info:

### Source code

- [x] I provided exact source code that allows reproducing the issue locally.

**Config file**

```js
// playwright.config.js
import { defineConfig, devices } from '@playwright/test';

export default defineConfig({
  projects: [
    {
      name: 'chromium',
      use: {
        ...devices['Desktop Chrome'],
      },
    },
    {
      name: 'firefox',
      use: {
        ...devices['Desktop Firefox'],
      },
    },
  ],
});
```

**Test file (self-contained)**

```js
import { test } from '@playwright/test';

test.use({
  locale: 'en-GB',
  timezoneId: 'Europe/London',
});

test('test-send-with-Accept-Language', async ({ page }) => {
  await page.evaluate(async () => {
    await fetch(
      "https://api.github.com/rate_limit", {
        headers: {
            'Content-Type': 'application/json',
            'Accept-Language': 'de'
        },
      });
  });
});
```

**Steps**

- Run the test with Chromium
- Look at the inspector network tab and check the sent headers
- See that Accept-Language=de is sent
- Run the test with Firefox
- Look at the inspector network tab and check the sent headers

**Expected**

- The request is sent with the Accept-Language=de header, same as with Chromium

**Actual**

- The request is sent with the Accept-Language=en-GB header instead

**Comment by yury-s** [comment]
What I see in 1.35.0 is that all three browsers send three different values. Chromium doesn't send accept-language at all unless you add `mode: 'no-cors'`and in that case the header is equal to `de`.

**Comment by bschelling** [comment]
I just ran into something very similar: The accept language set as a customheaders is ignored,  using the "locale" setting in the config works. I'd be fine if the "locale" setting would only overwrite the header if a value is present, but simply ignoring the accept-language header setting  without any warning is  evil.

**Comment by avetisk** [comment]
Indeed, I have the same issue where the custom `accept-language` header is overwritten in Firefox and Webkit. 

It's annoying as hell as this feature works fine with the other browsers.

I'm going to try @bschelling's suggestion and see if I can temporary solve this using the `locale` setting.

**Comment by mratzenb** [comment]
Are there any updates on this bug?  Version 1.46.1 has still the same behavior for Firefox.

**Comment by lo1tuma** [comment]
I’ve ran into the same issue. Is there anything I can help with in order to resolve this issue?

**Comment by NavaneethakannanPa** [comment]
I ran into the similar issue today. My request should have accept-language as 'zh' but it was overridden and set to 'en-US' which result 404. I manually overridden this config by setting `test.use({ locale: 'zh' });`. This helped me in this case. However any plan to fix this in the next version?.
